zerver/lib/test_classes.py

```diff
@@ -577,6 +579,22 @@ def client_get(
         )
         return result
 
+    def _prevent_refresh_from_db_for_user(self, user: UserProfile) -> None:
+        def fail() -> None:  # nocoverage
+            raise Exception(
+                """
+                Please do not call refresh_from_db on UserProfile
+                objects. Instead explicitly re-fetch the user object
+                from the database using a Zulip test helper such
+                as refresh_user or example_user.
+                """
+            )
```

Comment:
Let's add a description for why we're doing this -- maybe a new second sentence: "We want to avoid doing so because it is incompatible with using the django-seal library". or something like it.

Reply:
I think we can just drop this particular commit.  Once we have a culture to avoid `refresh_from_db` (I mean, we basically do already--pretty few uses of it in ten years), it won't be too much of a problem to catch it in code review.  And it's not a monumental thing, especially if we eventually have django-seal to reinforce the policy.